image/RasterImage.cpp

```diff
@@ -1358,9 +1366,27 @@ bool RasterImage::CanDownscaleDuringDecode(const UnorientedIntSize& aSize,
   // Check basic requirements: downscale-during-decode is enabled, Skia is
   // available, this image isn't transient, we have all the source data and know
   // our size, and the flags allow us to do it.
-  if (!LoadHasSize() || LoadTransient() || !HaveSkia() ||
-      !StaticPrefs::image_downscale_during_decode_enabled() ||
-      !(aFlags & imgIContainer::FLAG_HIGH_QUALITY_SCALING)) {
```

Comment:
Generally I think it's better if we try to leave the upstream code alone to as great a degree as possible.  This makes it easier to understand the changes we are making from the browser, to merge from upstream, and to remove assertions once they are no longer needed.

In this case that would be avoiding creating variables for the calls etc. in this test, and just repeating those calls in the RecordReplayAssert.  That's fine as long as the things being tested are side effect free, which looks to be the case here.